opm/parser/eclipse/EclipseState/Eclipse3DProperties.cpp

```diff
@@ -81,6 +99,57 @@ namespace Opm {
                 for (size_t globalIndex = 0; globalIndex < multpvData.size(); globalIndex++)
                     values[globalIndex] *= multpvData[globalIndex];
             }
+
+            // deal with the region multiplier for porosity
+            if (deck->hasKeyword("MULTREGP")) {
+                const DeckKeyword& multregpKeyword = deck->getKeyword("MULTREGP");
+                for (int recordIdx = 0; recordIdx < multregpKeyword.size(); ++recordIdx) {
+                    const DeckRecord& multregpRecord = multregpKeyword.getRecord(recordIdx);
+
+                    int regionIdx = multregpRecord.getItem("REGION").template get<int>(0);
+                    std::string regionType = multregpRecord.getItem("REGION_TYPE").template get<std::string>(0);
+                    double multValue = multregpRecord.getItem("MULTIPLIER").template get<double>(0);
+                    boost::to_upper(regionType);
+
+                    // deal with the "convenience" feature of ECL: if the region index is
+                    // zero or negative, the record is ignored. (why are such lines not
+                    // simply commented out? so that mere mortals do not understand
+                    // what's going on?)
+                    if (regionIdx <= 0)
+                        continue;
+
+                    // implement a (documented) ECL bug: the region index must be unique
```

Comment:
Again - the `MULTREGT` implementation implements the same quirks (I think) - would be good if the implementation could be shared.